tests/test_add_fund_and_claim.cairo

```diff
@@ -0,0 +1,418 @@
+    usdc_dispatcher.approve(salary_address, 150 * TOKEN_MULTIPLIER);
+    stop_prank(usdc);
+
+    let month_id = 092023;
+
+    let mut spy = spy_events(SpyOn::One(salary_address));
+    start_prank(salary_address, deployer_addr());
+    salary_dispatcher.add_fund_to_salary_pools(month_id, amounts, guilds);
+    stop_prank(salary_address);
+
+    // verifying balance of contract 
+    let contract_balance = usdc_dispatcher.balance_of(salary_address);
+    assert(contract_balance == 150 * TOKEN_MULTIPLIER, 'incorrect balance');
+
+    // defining variables to check in events emitted
+    let dev_guild = 'dev';
+    let dev_fund = 10 * TOKEN_MULTIPLIER;
+    let design_guild = 'design';
+    let design_fund = 20 * TOKEN_MULTIPLIER;
+    let problem_solving_guild = 'problem_solving';
+    let problem_solving_fund = 30 * TOKEN_MULTIPLIER;
+    let marcom_guild = 'marcom';
+    let marcom_fund = 40 * TOKEN_MULTIPLIER;
+    let research_guild = 'research';
+    let research_fund = 50 * TOKEN_MULTIPLIER;
+
+    let mut event_data_dev = Default::default();
+    Serde::serialize(@month_id, ref event_data_dev);
+    Serde::serialize(@dev_guild, ref event_data_dev);
+    Serde::serialize(@dev_fund, ref event_data_dev);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryPoolUpdated', keys: array![], data: event_data_dev }
+    ]);
+
+    let mut event_data_design = Default::default();
+    Serde::serialize(@month_id, ref event_data_design);
+    Serde::serialize(@design_guild, ref event_data_design);
+    Serde::serialize(@design_fund, ref event_data_design);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryPoolUpdated', keys: array![], data: event_data_design }
+    ]);
+
+    let mut event_data_problem_solving = Default::default();
+    Serde::serialize(@month_id, ref event_data_problem_solving);
+    Serde::serialize(@problem_solving_guild, ref event_data_problem_solving);
+    Serde::serialize(@problem_solving_fund, ref event_data_problem_solving);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryPoolUpdated', keys: array![], data: event_data_problem_solving }
+    ]);
+
+    let mut event_data_marcom = Default::default();
+    Serde::serialize(@month_id, ref event_data_marcom);
+    Serde::serialize(@marcom_guild, ref event_data_marcom);
+    Serde::serialize(@marcom_fund, ref event_data_marcom);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryPoolUpdated', keys: array![], data: event_data_marcom }
+    ]);
+
+    let mut event_data_research = Default::default();
+    Serde::serialize(@month_id, ref event_data_research);
+    Serde::serialize(@research_guild, ref event_data_research);
+    Serde::serialize(@research_fund, ref event_data_research);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryPoolUpdated', keys: array![], data: event_data_research }
+    ]);
+
+}
```

Comment:
Should the `_last_update_month_id` object value be checked?

Reply:
added